intercom-java/src/main/java/io/intercom/api/HttpClient.java

```diff
@@ -124,6 +110,10 @@ private HttpClient(URI uri, Map<String, String> headers) {
         }
     }
 
+    private <T> JavaType getJavaType(Class<T> reqres) {
+        return objectMapper.getTypeFactory().constructType(reqres);
+    }
```

Comment:
this lets us pass the type information through at runtime, which gets lost via erasure otherwise